### PR TITLE
Handling Breaking Changes in Frost API

### PR DESCRIPTION
**Pull request by lautarodragan**
Resolves https://github.com/poetapp/documentation/issues/30

### CODE REVIEW

**Comment by lautarodragan** [comment]
Read for review.

File: rdd/api-versioning.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+In the future we could do more research and put more thought on this topic. Some approaches to keep in mind:
+
+- Supporting an _endpoint feature_ system, rather than global or endpoint versioning.
+- For authenticated requests, storing the version of the API to use in the database for each user, allowing them to change it from an administration panel. 
+
+## Conclusion
+
+Following GitHub's approach, placing the version in the `Accept`header is the best option at the moment. It suits our current needs, requires no changes to the URLs, is already well documented and widely used.
+
+We will tag the current version as `v1`, and it will be the default initially and for a period of time. Once `v2` has been out for a while, the default will be switched to it, and so forth.
```
Comment (geoffturk) [inline rdd/api-versioning.md]: Current version is actually referred to as 0.1 from the beta period. Maybe we add the `claimId` in GET /works and release that as v1?

If no version is specified in the `Accept` header we default to 0.1.

Comment (lautarodragan) [reply rdd/api-versioning.md]: Good catch. Fixed now.

File: rdd/api-versioning.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- Version in Accept Header
+- Version in Custom Header
+
+## In the Wild
+
+Interesting read from the Have I Been Pwned author: [Your API versioning is wrong, which is why I decided to do it 3 different wrong ways](https://www.troyhunt.com/your-api-versioning-is-wrong-which-is).
+
+> 1. URLs suck because they should represent the entity: I actually kinda agree with this insofar as the entity I’m retrieving is a breached account, not a version of the breached account. Semantically, it’s not really correct but damn it’s easy to use!
+>
+> 1. Custom request headers suck because it’s not really a semantic way of describing the resource: The HTTP spec gives us a means of requesting the nature we’d like the resource represented in by way of the accept header, why reproduce this?
```
Comment (wzalazar) [inline rdd/api-versioning.md]: Should be the point 2?

File: rdd/api-versioning.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+> 
+> 1. Accept headers suck because they’re harder to test: I can no longer just give someone a URL and say “Here, click this”, rather they have to carefully construct the request and configure the accept header appropriately.
```
Comment (wzalazar) [inline rdd/api-versioning.md]: Should be the point 3?

Comment (kennylavender) [reply rdd/api-versioning.md]: He is using lazy markdown, when you view the file it increments the numbers automatically :)

Comment (kennylavender) [reply rdd/api-versioning.md]: Whether thats a bad or good thing, IDK. I've always numbered them manually myself.

Comment (geoffturk) [reply rdd/api-versioning.md]: It's bad when you have 20 items and then decide to remove one of them from the top of the list. ;-)

**Comment by wzalazar** [comment]
@lautarodragan awesome!!! 

how we will handle the breaking change the API with Frost Client, for example? any idea?

File: rdd/api-versioning.md (hunk 3 above)
Comment (wzalazar) [reply rdd/api-versioning.md]: oh, awesome, super lazy.... jaja

**Comment by lautarodragan** [comment]
@wzalazar good catch, I was forgetting about mentioning that in the doc. I'll update it :)

**Pull request merged**